### DISCUSSION TITLE
New collection community.ciscosmb

### DISCUSSION THREAD

**Discussion by qaxi** [ansible-collections/ansible-inclusion#26] [Resolved reviews]
Ansible module for Cisco SMB switches - SG300, SG500, SG350, SG550, CBS350
current pluggins - facts, command

Ansible Galaxy: https://galaxy.ansible.com/community/ciscosmb
GitHub: https://github.com/ansible-collections/community.ciscosmb
Issues Tracker: https://github.com/ansible-collections/community.ciscosmb/issues
GitHub Handles: @qaxi
Is the collection part of Automation Hub: No
We meet [Collection Requirements](https://github.com/ansible-collections/overview/blob/main/collection_requirements.rst): Yes

**Comment by tadeboro** [comment]
Thank you for your submission, @qaxi.

The items in the checklist can be in one of the following states:

  * Items with a checkmark and no attached comment are OK and need no intervention from your side.
  * Items with a checkmark that contain a comment will not block the inclusion process, but by following the advice given in the comment, the collection's quality can be improved.
  * Items with no checkmark and no extra details are waiting for evaluation.
  * Items with no checkmark that contain a comment need your attention.

Feel free to ask any questions and provide feedback. Thanks in advance for your understanding and patience.

_For details about the following points, refer to the [Collection Requirements](https://github.com/ansible-collections/overview/blob/main/collection_requirements.rst)._

Every comment should say whether the reviewer expects it to be addressed, or whether it's optional.

**Public availability and communication:**
- [x] published on Ansible Galaxy [Ansible Galaxy](https://galaxy.ansible.com) with version 1.0.0 or later
- [x] has a policy of releasing, versioning and deprecation announced to contributors and users in some way
- [x] have a Code of Conduct (CoC) compatible with the [Ansible Code of Conduct (CoC)](https://docs.ansible.com/ansible/latest/community/code_of_conduct.html)
- [x] have a publicly available issue tracker that does not require a paid level of service to create an account or view issues
- [x] have a public SCM repository, releases of the collection are tagged in this repository

**Standards and documentation:**
- [x] adheres to [semantic versioning](https://semver.org/)
- [x] follows [licensing rules](https://github.com/ansible-collections/overview/blob/main/collection_requirements.rst#licensing)
- [x] follows the [Ansible documentation standards](https://docs.ansible.com/ansible/devel/dev_guide/developing_modules_documenting.html) and the [style guide](https://docs.ansible.com/ansible/devel/dev_guide/style_guide/index.html#style-guide)
- [x] follows [development conventions](https://docs.ansible.com/ansible/devel/dev_guide/developing_modules_best_practices.html); as well as these other requirements:
  - [x] modules that only gather information are named `<something>_info`
  - [x] modules that return `ansible_facts` are named `<something>_facts` and do not return non-facts
  - [x] other modules must not allow querying information using specific `state` option values, or similar mechanisms (like `state=get` or `state=query`).  These features should be moved to `<something>_info` or `<something>_fact` modules.
  - [x] `check_mode` is supported in all `_info` and `_facts` modules
- [x] supports Python 2.6 or greater and Python 3.5 or greater. If it does not, read the [full guidelines](https://github.com/ansible-collections/overview/blob/main/collection_requirements.rst#python-compatibility) to see if you qualify for an exception and document the unsupported [Python versions](https://docs.ansible.com/ansible/latest/dev_guide/developing_python_3.html#ansible-and-python-3) in the collection ``README.md`` and in every module and plugin (or in doc fragments)
- [x] only uses the [allowed plugin types](https://github.com/ansible-collections/overview/blob/main/collection_requirements.rst#modules-plugins) in the `plugins/` directory
- [x] has `README.md`
- [x] documentation, examples, and return sections use FQCNs for the `M(..)` [format macros](https://docs.ansible.com/ansible/latest/dev_guide/developing_modules_documenting.html#linking-and-other-format-macros-within-module-documentation) when referring to modules
- [x] modules (or plugins) from ansible-core use `ansible.builtin.` as a FQCN prefix
- [x] documentation and return sections use `version_added:` containing the *collection* version for which an option, module or plugin was added
- [x] FQCNs are used for `extends_documentation_fragment:`, unless you are referring to doc_fragments from ansible-core

**Collection management:**
- [x] `galaxy.yml` having `tags` field set
- [x] collection dependencies must have a lower bound on the version which is at least 1.0.0, and are all part of the `ansible` package
- [x] `meta/runtime.yml` defines the minimal version of Ansible which the collection works with
- [x] has changelog, preferably with `changelogs/changelog.yaml`

**Tests:**
- [x] passed `ansible-test sanity`
- [x] if `test/sanity/ignore*.txt` exists, it does not contain error codes listed [here](https://github.com/ansible-collections/overview/blob/main/collection_requirements.rst#ci-testing)
- [x] has CI tests up and running against each of the "major versions" of `ansible-base`/`ansible-core` that the collection supports
- [x] all CI tests run against every pull request
- [x] all CI tests run against a commit that releases the collection; if they don't pass, the collection won't be released
- [x] all CI tests run regularly (nightly, or at least once per week)

**Reply by tadeboro** [reply]
I finished my initial review. Collection is currently not ready to be included in the `ansible` package.

**Reply by qaxi** [reply]
Hello Tadej,

I fixed all problem parts in commit https://github.com/ansible-collections/community.ciscosmb/commit/e0080406cbf833cbd8a18dbb9d9a5640b4fff4b5

> 
> **Public availability and communication:**
> 
>     * [ ]  has a policy of releasing, versioning and deprecation announced to contributors and users in some way
>       **MUST FIX:** The collection documentation does not contain this information. See [community.sops](https://github.com/ansible-collections/community.sops#releasing-versioning-and-deprecation), [dellemc.sonicenterprise](https://github.com/ansible-collections/dellemc.enterprise_sonic#releasing-versioning-and-deprecation), or [sensu.sensu_go](https://sensu.github.io/sensu-go-ansible/release_policy.html) for example.
> 
>     * [ ]  have a Code of Conduct (CoC) compatible with the [Ansible Code of Conduct (CoC)](https://docs.ansible.com/ansible/latest/community/code_of_conduct.html)
>       **MUST FIX:** The enforcement section does not contain a valid email address as required by the Contributor Covenant v1.4.
>       **MUST FIX:** The CoC text is a bit out of sync with the source at https://github.com/EthicalSource/contributor_covenant/blob/release/content/version/1/4/code-of-conduct.md
>       **COMMENT:** It would be nice to have a link to CoC in the readme.
> 
> 
> **Standards and documentation:**
> 
>     * [ ]  adheres to [semantic versioning](https://semver.org/)
>       **MUST FIX:** The version numbers do match the semver regex, but it is hard to tell is the collection does follow semver rules without having a document describing release policies.
> 
>     * [ ]  follows [licensing rules](https://github.com/ansible-collections/overview/blob/main/collection_requirements.rst#licensing)
>       **MUST FIX:** https://github.com/ansible-collections/community.ciscosmb/blob/main/plugins/module_utils/ciscosmb_canonical_map.py is missing a license header.
> 
>     * [ ]  follows the [Ansible documentation standards](https://docs.ansible.com/ansible/devel/dev_guide/developing_modules_documenting.html) and the [style guide](https://docs.ansible.com/ansible/devel/dev_guide/style_guide/index.html#style-guide)
>       **MUST FIX:** https://github.com/ansible-collections/community.ciscosmb/blob/main/plugins/cliconf/ciscosmb.py#L24 is missing an author.
>

**Reply by tadeboro** [reply]
Hi @qaxi! Thank you for your quick response. Things are looking great, I just found one minor copy-paste error in the readme file. Once that is fixed and a new release with the changes from this review is uploaded to Ansible Galaxy, the `community.ciscosmb` is ready for inclusion as far as I am concerned.

**Reply by qaxi** [reply]
Hi @tadeboro,
I am sorry, I can not find any error in README.md. Can you give me a hint?

**Reply by tadeboro** [reply]
https://github.com/ansible-collections/community.ciscosmb/blob/main/README.md#code-of-conduct section contains a sentence that mentions Ansible's CoC instead of contributor covenant.

**Reply by qaxi** [reply]
Thank you ...
fixed by https://github.com/ansible-collections/community.ciscosmb/pull/36/commits/ef3bfa534c06149b4ee716259e42fac473dc30b3

**Reply by tadeboro** [reply]
Thank you, @qaxi, for your cooperation. As far as I am concerned, the `community.ciscosmb` is ready for inclusion. Changes made during the review process should be released before Ansible 5.0.0 is out, but since there is a lot of time remaining until that, you can safely wait for another regular release (no need to create a dedicated patch release just for the sake of inclusion).

**Comment by Andersson007** [comment]
**Public availability and communication:**
- [x] published on Ansible Galaxy [Ansible Galaxy](https://galaxy.ansible.com) with version 1.0.0 or later
- [x] has a policy of releasing, versioning and deprecation announced to contributors and users in some way
- [x] have a Code of Conduct (CoC) compatible with the [Ansible Code of Conduct (CoC)](https://docs.ansible.com/ansible/latest/community/code_of_conduct.html)
- [x] have a publicly available issue tracker that does not require a paid level of service to create an account or view issues
- [x] have a public SCM repository, releases of the collection are tagged in this repository

**Standards and documentation:**
- [x] adheres to [semantic versioning](https://semver.org/)
- [x] follows [licensing rules](https://github.com/ansible-collections/overview/blob/main/collection_requirements.rst#licensing)
- [x] follows the [Ansible documentation standards](https://docs.ansible.com/ansible/devel/dev_guide/developing_modules_documenting.html) and the [style guide](https://docs.ansible.com/ansible/devel/dev_guide/style_guide/index.html#style-guide)
  **REQUIRED (DONE)**: merging https://github.com/ansible-collections/community.ciscosmb/pull/41
- [x] follows [development conventions](https://docs.ansible.com/ansible/devel/dev_guide/developing_modules_best_practices.html); as well as these other requirements:
  - [x] modules that only gather information are named `<something>_info`
  - [x] modules that return `ansible_facts` are named `<something>_facts` and do not return non-facts
  - [x] other modules must not allow querying information using specific `state` option values, or similar mechanisms (like `state=get` or `state=query`).  These features should be moved to `<something>_info` or `<something>_fact` modules.
  - [x] `check_mode` is supported in all `_info` and `_facts` modules
- [x] supports Python 2.6 or greater and Python 3.5 or greater. If it does not, read the [full guidelines](https://github.com/ansible-collections/overview/blob/main/collection_requirements.rst#python-compatibility) to see if you qualify for an exception and document the unsupported [Python versions](https://docs.ansible.com/ansible/latest/dev_guide/developing_python_3.html#ansible-and-python-3) in the collection ``README.md`` and in every module and plugin (or in doc fragments)
  **REQUIRED**: would be nice to see supported Python versions in `README.md`
- [x] only uses the [allowed plugin types](https://github.com/ansible-collections/overview/blob/main/collection_requirements.rst#modules-plugins) in the `plugins/` directory
- [x] has `README.md`
- [x] documentation, examples, and return sections use FQCNs for the `M(..)` [format macros](https://docs.ansible.com/ansible/latest/dev_guide/developing_modules_documenting.html#linking-and-other-format-macros-within-module-documentation) when referring to modules
- [x] modules (or plugins) from ansible-core use `ansible.builtin.` as a FQCN prefix
- [x] documentation and return sections use `version_added:` containing the *collection* version for which an option, module or plugin was added
- [x] FQCNs are used for `extends_documentation_fragment:`, unless you are referring to doc_fragments from ansible-core

**Collection management:**
- [x] `galaxy.yml` having `tags` field set
  **REQUIRED:** merging https://github.com/ansible-collections/community.ciscosmb/pull/42
- [x] collection dependencies must have a lower bound on the version which is at least 1.0.0, and are all part of the `ansible` package
- [x] `meta/runtime.yml` defines the minimal version of Ansible which the collection works with
- [x] has changelog, preferably with `changelogs/changelog.yaml`
  **RECOMMENDATION:** putting links to issues/PRs to fragments so that users will find details easily if needed.

**Tests:**
- [x] passed `ansible-test sanity`
- [x] if `test/sanity/ignore*.txt` exists, it does not contain error codes listed [here](https://github.com/ansible-collections/overview/blob/main/collection_requirements.rst#ci-testing)
- [x] has CI tests up and running against each of the "major versions" of `ansible-base`/`ansible-core` that the collection supports
- [x] all CI tests run against every pull request
- [x] all CI tests run against a commit that releases the collection; if they don't pass, the collection won't be released
- [x] all CI tests run regularly (nightly, or at least once per week)
  REQUIRED: Clarifications on https://github.com/ansible-collections/ansible-inclusion/discussions/26#discussioncomment-1306742

**Reply by Andersson007** [reply]
@qaxi i see the `command` module `supports_check_mode=True` and it's just not considered anywhere.
Can the module not only query devices but change their state? If yes, this can lead to unexpected consequences for users who expect that nothing will change when they invoke the module with `check_mode: yes`.

**Reply by Andersson007** [reply]
I see in the GH test matrix that CI runs against the following Python versions:
```
 94         python:
 95           - 2.6
 96           - 2.7
 97           - 3.7
 98           - 3.8
 99           - 3.9
```
In the [requirements](https://github.com/ansible-collections/overview/blob/main/collection_requirements.rst#python-requirements) said:
```
Controller-environment

In the controller environment, collections MUST support Python 2 (version 2.7) and Python 3 (Version 3.6 and higher), unless required libraries do not support these versions. Collections SHOULD also support Python v3.5 if all required libraries support this version.

Other-environment

In the other environment, collections MUST support Python 2 (version 2.7) and Python 3 (Version 3.6 and higher), unless required libraries do not support these versions. Collections SHOULD also support Python v2.6 and v3.5 if all required libraries support this version.
```

Can we add **Python 3.6** to `the matrix` and mention all supported versions in `README.md`?

**Reply by qaxi** [reply]
OK I will add 3.6 in test branch and alter documetation in 15 minutes ...

**Reply by Andersson007** [reply]
Cool, thanks! Also we should decide something on https://github.com/ansible-collections/ansible-inclusion/discussions/26#discussioncomment-1306469

**Reply by qaxi** [reply]
Fixed: Add Py 3.6 to supported python versions #44

**Reply by qaxi** [reply]
galaxy.xml fix merged [#43](https://github.com/ansible-collections/community.ciscosmb/pull/43)
Misc doc fixes for collection inclusion merged [#41](https://github.com/ansible-collections/community.ciscosmb/pull/41)

**Reply by qaxi** [reply]
The changes are published in release 1.0.3

**Reply by qaxi** [reply]
> @qaxi i see the `command` module `supports_check_mode=True` and it's just not considered anywhere.
> Can the module not only query devices but change their state? If yes, this can lead to unexpected consequences for users who expect that nothing will change when they invoke the module with `check_mode: yes`.

... it definitely does not support_check_mode ... PR [#45](https://github.com/ansible-collections/community.ciscosmb/pull/45)

will release ASAP

**Reply by Andersson007** [reply]
@qaxi thanks for the fast response and fixing all the things in the collection!
+1 from me to its inclusion

**Comment by felixfontein** [comment]
This collection will be included in Ansible 5: https://github.com/ansible-community/ansible-build-data/pull/70
